src/kms/DeterministicCrypto.ts

```diff
@@ -0,0 +1,227 @@
+import Future from "futurejs";
+import {TenantSecurityErrorCode, TenantSecurityException} from "../TenantSecurityException";
+import {TenantSecurityExceptionUtils} from "../TenantSecurityExceptionUtils";
+import {TscException} from "../TscException";
+import * as miscreant from "miscreant";
+import {
+    BatchResult,
+    DeterministicEncryptedField,
+    DeterministicEncryptedFieldCollection,
+    DeterministicPlaintextField,
+    DeterministicPlaintextFieldCollection,
+    mapBatchOperationToResult,
+} from "../Util";
+import {DETERMINISTIC_HEADER_PADDING} from "./Constants";
+import {DerivedKey, DeriveKeyResponse, getDerivedKeys} from "./KmsApi";
+
+const cryptoProvider = new miscreant.PolyfillCryptoProvider();
+
+interface DeterministicEncryptedFieldParts {
+    tenantSecretId: number;
+    encryptedBytes: Buffer;
+}
+
+export type BatchDeterministicEncryptResult = TenantSecurityException | DeterministicEncryptedField;
+export type BatchDeterministicDecryptResult = TenantSecurityException | DeterministicPlaintextField;
+export type BatchDeterministicSearchResult = TenantSecurityException | DeterministicEncryptedField[];
+
+/**
+ * Deterministically encrypt the provided field with the current key from `derivedKeys`.
+ */
+export const encryptField = (
+    field: DeterministicPlaintextField,
+    derivedKeys: DerivedKey[] | undefined
+): Future<TenantSecurityException, DeterministicEncryptedField> => {
+    const current = derivedKeys?.find((key) => key.current);
+    if (current === undefined) {
+        return Future.reject(new TscException(TenantSecurityErrorCode.DETERMINISTIC_FIELD_ENCRYPT_FAILED, "Failed deterministic encryption."));
+    }
+    const dekBytes = Buffer.from(current.derivedKey, "base64");
```

Comment:
What happens if the derivedKey isn't base64?

Reply:
I added a test to show that the encrypt will fail